spec.bs

```diff
@@ -0,0 +1,369 @@
+
+      // Get data for an ad creative.
+      const response = await fetch('https://ads.example/get-creative', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(topics)
+      });
+
+      // Get the JSON from the response.
+      const creative = await response.json();
+
+      // Display the ad.
+    </pre>
+  </div>
+
+  <div class="example">
+    Creating an ad based on the browsing interests, based on the Sec-Browsing-Topics navigation request header:
+
+    (Inside the top level context)
+    <pre class="lang-js">
+      // A 'Sec-Browsing-Topics: [topics header value]' header will be sent in
+      // the HTTP request.
+      &ltiframe src="https://ads.example/get-creative"&gt&lt/iframe&gt
+    </pre>
+  </div>
+
+  <div class="example">
+    Creating an ad based on the browsing interests, based on the Sec-Browsing-Topics fetch() request header:
+
+    (Inside the top level context)
+    <pre class="lang-js">
+      // A 'Sec-Browsing-Topics: [topics header value]' header will be sent in
+      // the HTTP request.
+      const response = await fetch('https://ads.example/get-creative', {browsingTopics: true});
+      const ad_creative = await response.json();
+      // Display the ad.
+    </pre>
+  </div>
+</section>
+
+<section>
+  <h2 id="browsing-topic-dictionary-header">BrowsingTopic dictionary</h2>
+  The {{BrowsingTopic}} dictionary is used to contain the IDL correspondences of [=topic id=], [=version=], [=configuration version=], [=taxonomy version=], and [=model version=].
+
+  <pre class="idl">
+  dictionary BrowsingTopic {
+    long topic;
+    DOMString version;
+    DOMString configVersion;
+    DOMString modelVersion;
+    DOMString taxonomyVersion;
+  };
+  </pre>
+
+  A <dfn>taxonomy</dfn> comprises a list of advertising <dfn>topics</dfn>. A [=taxonomy=] is identified by a <dfn>taxonomy version</dfn>, and a [=topic=] is identified by a <dfn>topic id</dfn>.
+
+  Note: The <a href="https://github.com/patcg-individual-drafts/topics/blob/main/taxonomy_v1.md">initial taxonomy</a> for Chrome experimentation includes hundreds of topics. The taxonomy to be used long term is TBD (<a href="https://github.com/patcg-individual-drafts/topics/issues/3">github issue</a>). 
+
+  The <dfn>model version</dfn> identifies the model used to classify a page into a [=topic=]. The meaning may vary across browser vendors.
+
+  The <dfn>configuration version</dfn> identifies the algorithm (other than the model part) used to calculate the topic. The meaning may vary across browser vendors.
+
+  The <dfn>version</dfn> is the concatenation of browser vendor name, [=configuration version=], [=taxonomy version=], and [=model version=].
+
+  <div class="example">
+  In Chrome, the [=version=] takes the form of <code>chrome.[=configuration version=]:[=taxonomy version=]:[=model version=]</code>.
+  <div>
+
+</section>
+
+<section>
```

Comment:
This section is not sufficiently precise, and mostly needs a rewrite. It uses too many undefined concepts, like "top level page", "the document", "fetch() initiator Window", "navigation request", "happens on an iframe", "top level page associated with the iframe", etc.

Instead, you need to break it down as follows:

- A description of the "topics caller context" struct (nit: I suggest removing the hyphens), with its two fields, "caller origin" and "page".
   - State the "caller origin" is a `[=origin=]`
   - Give "page" a proper type, maybe `{{Document}}`? And probably rename it to something better.
   - Make sure their definitions have `for="topics caller context"`
- Two algorithms for rigorously creating a "topics caller context", with its two fields filled in. One for the `document.browsingTopics()` method to call, and one for the `fetch()` spec to call.
  - It looks like the first should take as input a `{{Document}}`, from which you can more-rigorously look at the fields of (via cross-references, like `|document|'s [=Document/origin=]` or `|document|'s [=top-level browsing context=]'s [=active document=]`)).
  - The latter should take as input a `[=request=]`, which you can look at other fields of.
- Update the appropriate parts of the spec to call these two algorithms as necessary.
- Update all references to caller-origin and page that are used in this way to be appropriate cross-links, e.g. `[=topics caller context/caller origin=]`.